dev/io.openliberty.mail.2.1.internal_fat/fat/src/io/openliberty/mail/fat/StreamProviderTest.java

```diff
@@ -0,0 +1,228 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.mail.fat;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.ProtocolException;
+import java.net.URL;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.custom.junit.runner.Mode;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.impl.LibertyServerFactory;
+
+@Mode(TestMode.FULL)
+@RunWith(FATRunner.class)
+public class StreamProviderTest {
+    private static LibertyServer server = LibertyServerFactory.getLibertyServer("mailSessionTestServer");
+    private final Class<?> c = StreamProviderTest.class;
+
+    private static String stringURL = null;
+    private static boolean streamProviderTestPassed = false;
+    private static final String STREAM_PROVIDER_ERROR_MESSAGE = "Failed to get stream provider from session. This test won't get executed.";
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        if (server.isStarted() != true) {
+            server.startServer();
+            // Pause for application to start properly and server to say it's listening on ports
+            server.waitForStringInLog("port " + server.getHttpDefaultPort());
+        }
+        stringURL = "http://" + server.getHostname() + ":" + server.getHttpDefaultPort() + "/TestingApp/StreamProviderSessionServlet";
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        if (server != null && server.isStarted()) {
+            server.stopServer("CWWKZ0013E");
+        }
+    }
+
+    /**
+     * TestDescription:
+     *
+     * Tests newly added StreamProvider class to Jakarta Mail 2.1 by invoking getStreamProvider method
+     * from Session class in StreamProviderSessionServlet servlet.
+     *
+     * This test is essential for all other tests to run.
+     */
+    @Test
```

Comment:
Now that I'm looking at the code, I don't think this will work, since we can't depend on test order in our test framework. Sorry, I should have realized this is what you meant earlier. What you can do instead, is just make the setup method call this method directly, so it gets run before anything else in the test. Something like this: 

```
    @BeforeClass
    public static void setup() throws Exception {
        if (server.isStarted() != true) {
            server.startServer();
            // Pause for application to start properly and server to say it's listening on ports
            server.waitForStringInLog("port " + server.getHttpDefaultPort());
        }
        stringURL = "http://" + server.getHostname() + ":" + server.getHttpDefaultPort() + "/TestingApp/StreamProviderSessionServlet";

        testNewStreamProvider()
    }
```

Then just remove the `@Test` annotation from this method.

Reply:
I redesigned the tests to overcome this issue.